### PR TITLE
Fix classic UT failing test cases after update to 7.4.3

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

* Fix classic UT failing test cases after update to 7.4.3

### Description

* Fix classic UT failing test cases after update to 7.4.3

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by biaol-odata** [comment]
FYI: The following PR contains lots of changes on WebAPI/master branch upgrading ODL dependency 7.3.1->7.4.1, and might save some of your time fixing the failing tests. (I saw some of them here, but not sure you have them all).

**Comment by biaol-odata** [comment]
https://github.com/OData/WebApi/pull/1225

**Comment by xuzhg** [comment]
@biaol-odata Thanks! That should be helpful. However, all classic failing test cases are included here.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Builder/ContainmentPathBuilderTest.cs [hunk 1]
```diff
@@ -49,9 +49,9 @@ public class ContainmentPathBuilderTest
         [InlineData("Clubs/ns.SeniorClub(1)/Members(1)/ns.VIP/MyBenefits", "Clubs(1)/Members(1)/ns.VIP")]
         [InlineData("Clubs/ns.SeniorClub(1)/Members/ns.VIP(1)/MyBenefits", "Clubs(1)/Members(1)/ns.VIP")]
         // With redundant EntitySet path
-        [InlineData("Clubs(1)/ns.SeniorClub/Members(1)/ns.VIP/MyAccounts(101)/MyPaymentInstruments", "Clubs(1)/Members(1)/MyAccounts(101)")]
-        [InlineData("Clubs/ns.SeniorClub(1)/Members(1)/ns.VIP/MyAccounts(101)/MyPaymentInstruments", "Clubs(1)/Members(1)/MyAccounts(101)")]
-        [InlineData("Clubs/ns.SeniorClub(1)/Members/ns.VIP(1)/MyAccounts(101)/MyPaymentInstruments", "Clubs(1)/Members(1)/MyAccounts(101)")]
+        [InlineData("Clubs(1)/ns.SeniorClub/Members(1)/ns.VIP/MyAccounts(101)/MyPaymentInstruments", "Accounts(101)")]
```
Comment (biaol-odata) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Builder/ContainmentPathBuilderTest.cs]: Q: Are we making containedEntitySet directly addressable thru top-level entity-set? But direct access is identified as an issue from this blog https://blogs.msdn.microsoft.com/odatateam/2014/03/12/tutorial-sample-containment-is-coming-with-odata-v4/.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Query/FilterQueryOptionTest.cs [hunk 2]
```diff
@@ -769,7 +770,6 @@ public void ApplyWithCast_ReturnsCorrectQueryable(string filter, int[] castModel
         [InlineData("cast(NoSuchProperty,Edm.String) eq null", typeof(ODataException))]
         [InlineData("cast(ProductId,Edm.NoSuchType) eq null", typeof(ODataException))]
         [InlineData("cast(ProductId,Edm.String) eq 123", typeof(ODataException))]
-        [InlineData("cast('123',Microsoft.Test.AspNet.OData.Common.Types.SimpleEnum) ne 'First'", typeof(ODataException))]
```
Comment (biaol-odata) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Query/FilterQueryOptionTest.cs]: Not sure why we are removing this test case. Is there a way to keep it and update the expected data of exceptionType properly instead?

Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/Query/FilterQueryOptionTest.cs]: Not removed. Now, it's a valid case. Move it to the valid test case.
#Resolved.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Builder/ContainmentPathBuilderTest.cs (hunk 1 above)
Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/Builder/ContainmentPathBuilderTest.cs]: No. I think that's the change happened in ODL 7.4.*.

**Comment by biaol-odata** [comment]
Current understanding is that this is WIP... looks good to me so far.

**Pull request merged**